torch/tensor.py

```diff
@@ -376,6 +376,12 @@ def split(self, split_size, dim=0):
         """
         if isinstance(split_size, int):
             return super(Tensor, self).split(split_size, dim)
+        elif isinstance(split_size, Tensor):
+            try:
+                split_size = int(split_size)
+                return super(Tensor, self).split(split_size, dim)
+            except ValueError:
+                return super(Tensor, self).split_with_sizes(split_size, dim)
```

Comment:
Does `split_with_sizes` support `split_size` of type `Tensor`?

Reply:
No it does not support a tensor as of now, but the idea is that if the tensor is not actually a scalar, it should fail in the same way that it currently does.